src/react-components/src/components/Link/Link.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from "react";
+import bem from "../../utils/bem";
+import { uid } from "react-uid";
+import { LinkTypes } from "./LinkTypes";
+import { element } from "prop-types";
+import Icon from "../Icons/Icon";
+import { iconRotationTypes } from "../Icons/IconTypes";
+
+export interface LinkProps {
+    /** Controls the link visuals—action, button, or default. */
+    linkType?: LinkTypes;
+
+    /** Href attribute */
+    href?: string;
+
+    /** BEM specifiers */
+    modifiers?: string[];
+
+    /** BEM specifiers */
+    blockName?: string;
+
+    /** Additional attributes, such as rel=nofollow, to pass to the <a> tag */
+    attributes?: {};
+}
+
+export default function Link(props: React.PropsWithChildren<LinkProps>) {
+    const {
+        href,
+        modifiers,
+        blockName,
+        attributes,
+        linkType = LinkTypes.Default,
+    } = props;
+
+    const passedProps = { href, attributes };
+    passedProps.href = props.href;
+    passedProps.attributes = props.attributes;
+
+    let link_base_class = "link";
+
+    let childProps = {};
+
+    let icon;
+
+    let elementChildren = React.Children.map(
+        props.children,
+        (child: React.ReactElement) => {
+            if (typeof child === "string" && !props.href) {
+                throw new Error("Link needs prop 'href'");
+            } else if (child.type === "a") {
+                childProps = child.props;
+                return child.props.children;
+            } else {
+                return child;
+            }
+        }
+    );
+
+    if (
+        props.linkType === LinkTypes.Action ||
```

Comment:
You should not need to call `props` within the body of the function at all, because this variable is set on L32.  
(This isn't as subtle as it seems, forcing the `const` on it so we always know whether or not it's been changed is one of the great benefits of TS)

Reply:
I sorta figured once it worked in other places... thanks for catching my messes 😭